apps/client/src/app/services/internet-identity.service.ts

```diff
@@ -0,0 +1,42 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { AuthClient } from '@dfinity/auth-client';
+import { OAuthResponse } from '@ghostfolio/common/interfaces';
+import { EMPTY } from 'rxjs';
+import { catchError } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class InternetIdentityService {
+  public constructor(private http: HttpClient) {}
+
+  public async login(): Promise<OAuthResponse> {
+    const authClient = await AuthClient.create();
+
+    return await new Promise((resolve, reject) => {
+      authClient.login({
```

Comment:
Here also, the default delegation least 10min (I think or something).

If you wish to defined a longer session you can do as following:

```
// How long the delegation identity should remain valid?
// e.g. BigInt(7 * 24 * 60 * 60 * 1000 * 1000 * 1000) = 7 days in nanoseconds
// For Papyrs: 4 hours
const delegationIdentityExpiration: bigint = BigInt(4 * 60 * 60 * 1000 * 1000 * 1000);

await authClient.login({
    maxTimeToLive: delegationIdentityExpiration,
    ...
```

Reply:
See above.